src/platform/i386/lapic.c

```diff
@@ -357,10 +404,99 @@ lapic_timer_init(void)
 
 	/* Set the divisor */
 	lapic_write_reg(LAPIC_DIV_CONF_REG, LAPIC_DIV_BY_1);
+
+	lapic_ack();
+}
+
+static int
+lapic_ipi_send(u32_t dest, u32_t vect_flags)
+{
+	lapic_write_reg(LAPIC_ICR + 0x10, dest << 24);
+	lapic_read_reg(LAPIC_ICR + 0x10);
+
+	lapic_write_reg(LAPIC_ICR, vect_flags);
+	lapic_read_reg(LAPIC_ICR);
+
+	return 0;
+}
+
+/* HACK: assume that the HZ of the processor is equivalent to that on the computer used for compilation. */
```

Comment:
Can we use the clock speed returned by the msr?

I don't have a good solution to this.  @phanikishoreg ?

Reply:
That seems like a great idea to me. We should use the clock speed from the MSR if it's available and fall back to a HACKish solution if it's not.

Reply:
Now we have a global variable to get the MSR frequency.. So you should use that if it's not zero!